src/sd/sd_client.cpp

```
@@ -207,6 +212,7 @@ class SdClientImpl : public transport::ITransportListener {
         endpoint_option->set_protocol(0x11);  // UDP
         sd_message.add_option(std::move(endpoint_option));
 
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-static-cast-downcast)
         auto* sub_entry = static_cast<EventGroupEntry*>(sd_message.get_entries()[0].get());
         sub_entry->set_index1(0);
```

Cppcheck (2.20.0)

[style] The function 'get_flags' is never used.

(unusedFunction)